### PR TITLE
Closes #3578: Api for excluding the selectedSession on the SessionSuggestionProvider.

### LINKED ISSUE #3578: SessionSuggestionProvider shows current selected session, does nothing
See https://github.com/mozilla-mobile/fenix/issues/3695

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SessionSuggestionProvider.kt [hunk 1]
```diff
@@ -48,4 +49,17 @@ class SessionSuggestionProvider(
 
         return suggestions
     }
+
+    private fun Session.contains(text: String) =
+        (url.contains(text, ignoreCase = true) || title.contains(text, ignoreCase = true))
+
+    private fun shouldIncludeSelectedSession(session: Session): Boolean {
+        return if (excludeSelectedSession) {
+            !(isSelectedSession(session) ?: false)
```
Comment (pocmo) [inline components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SessionSuggestionProvider.kt]: nit: this is pretty hard to read. `isSelectedSession` probably should just return a non-null boolean.

Comment (Amejia481) [reply components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SessionSuggestionProvider.kt]: Updated 👍

File: docs/changelog.md [hunk 2]
```diff
@@ -12,6 +12,12 @@ permalink: /changelog/
 * [Gecko](https://github.com/mozilla-mobile/android-components/blob/master/buildSrc/src/main/java/Gecko.kt)
 * [Configuration](https://github.com/mozilla-mobile/android-components/blob/master/buildSrc/src/main/java/Config.kt)
 
+* **feature-tab**
+  * ⚠️ **This is a breaking change**: Now `TabsUseCases.SelectTabUseCase` is an interface, if you want to use is previous behavior you could keep using  `TabsUseCases.selectTab` or us `TabsUseCases.DefaultSelectTabUseCase`
+
+* **feature-awesomebar**
+  * ⚠️ **This is a breaking change**: `SessionSuggestionProvider` now have a new parameter `excludeSelectedSession`, to ignore the selected session on the suggestions.
```
Comment (csadilek) [inline docs/changelog.md]: nit: This shouldn't be a breaking change as `excludeSelectedSession` defaults to false, which is the same as before?

Comment (Amejia481) [reply docs/changelog.md]: Thanks for the catch, updated!

**Comment by ekager** [comment]
I may be confused but from my understanding this still won't fix Fenix's problem. Clicking the currently selected session will never do anything in Fenix because this lives in SearchFragment. I think we need a custom use case here to be able to hook into this onSuggestionClicked action and route the user to BrowserFragment.

**Comment by Amejia481** [comment]
> I may be confused but from my understanding this still won't fix Fenix's problem. Clicking the currently selected session will never do anything in Fenix because this lives in SearchFragment. I think we need a custom use case here to be able to hook into this onSuggestionClicked action and route the user to BrowserFragment.

Now that `TabsUseCases.SelectTabUseCase` is an interface you can provide your own implementation use case, using the same pattern as you are doing with other suggestion providers like [loadUrlUseCase](https://github.com/mozilla-mobile/fenix/blob/867067537a7dccf3d5f17dfbf3730f076cbc4bc2/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarUIView.kt#L72), [searchUseCase](https://github.com/mozilla-mobile/fenix/blob/867067537a7dccf3d5f17dfbf3730f076cbc4bc2/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarUIView.kt#L78) and [shortcutSearchUseCase](https://github.com/mozilla-mobile/fenix/blob/867067537a7dccf3d5f17dfbf3730f076cbc4bc2/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarUIView.kt#L84) and get hooked when the item is selected.

Something like: 

```kotlin
    private val selectTapUseCase = object : TabsUseCases.SelectTabUseCase {
        override fun invoke(session: Session) {
            actionEmitter.onNext(AwesomeBarAction.URLTapped(session.url))
        }
    }
```

and when you init the `sessionProvider`:

```kotlin

            sessionProvider =
                SessionSuggestionProvider(
                    components.core.sessionManager,
                    selectTapUseCase,
                    components.core.icons,
                    excludeSelectedSession = true
                )
```

Additionally with the new property `excludeSelectedSession`,  now you can control if the selected session should be included in the suggestion list.

For a full example take a look [here](https://github.com/Amejia481/fenix/commit/0182f30f61ba7e07b78b821dc4b96e03e588aa86). Please play with it locally, and let me know if there is any point of improvements :)

**Comment by ekager** [comment]
Okay! I think this will work for us then! I think UX still needs to decide if/when you should see your current session, but it will be useful to have this available :)

File: docs/changelog.md [hunk 3]
```diff
@@ -56,6 +56,12 @@ permalink: /changelog/
 * **feature-media**
   * Added `MediaNotificationFeature` - a feature implementation to show an ongoing notification (keeping the app process alive) while web content is playing media.
 
+* **feature-tab**
+  * ⚠️ **This is a breaking change**: Now `TabsUseCases.SelectTabUseCase` is an interface, if you want to use is previous behavior you could keep using  `TabsUseCases.selectTab` or use `TabsUseCases.DefaultSelectTabUseCase`
```
Comment (csadilek) [inline docs/changelog.md]: nit: want to rely on its...

Comment (Amejia481) [reply docs/changelog.md]: Thanks for the catch, updated!

**Pull request merged**